src/internal/ppsdb/ppsdb.go

```diff
@@ -17,7 +19,18 @@ const (
 	jobsCollectionName      = "jobs"
 )
 
-var pipelinesIndexes = []*col.Index{}
+// PipelinesVersionIndex records the version numbers of pipelines
+var PipelinesVersionIndex = &col.Index{
+	Name: "version",
+	Extract: func(val proto.Message) string {
+		info := val.(*pps.PipelineInfo)
+		return fmt.Sprintf("%s@%d", info.Pipeline.Name, info.Version)
```

Comment:
The number needs to 0 padded for it to be lexicographically sorted.

Reply:
Ah true, I don't think I'm relying on that, but couldn't hurt. 9999 pipeline versions ought to be enough for anyone

Reply:
I think we should bump the padding a bit higher. 9999 versions is probably fine in the majority of cases, but there isn't really a downside to bumping this higher. I think 8 would be high enough to make it never be an issue.